### PR TITLE
feat(SelectCountry): use autocomplete instead of dropdown

### LINKED ISSUE #2993: feat(SelectCountry): return the full country object when selecting a country
## 🚀 Feature Proposal

As of today, we return the ISO value of the [country object](https://github.com/dnbexperience/eufemia/blob/main/packages/dnb-eufemia/src/extensions/forms/constants/countries.ts) for the selected country.
Could it be smarter to return the full object? As users might want to get different data/value/keys/properties instead of the ISO value only.

Docs and examples for the SelectCountry object can be seen [here](https://eufemia.dnb.no/uilib/extensions/forms/feature-fields/SelectCountry/demos/#empty).

### PR DESCRIPTION
**Pull request by tujoworker**
With that change we support browser auto-fill and much quicker UX search ability. Also, users can search in both/all locales at the same time.

Fixes #2993

### CODE REVIEW

**Review by joakbjerk — APPROVED** [review]
Looks good! 🫶

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/SelectCountry/properties.mdx [hunk 1]
```diff
@@ -6,9 +6,10 @@ import DataValueReadwriteProperties from '../../data-value-readwrite-properties.
...
-| --------------------------------------- | -------- | -------------------------------------------------------------------------------- |
-| `help`                                  | `object` | _(optional)_ Provide a help button. Object consisting of `title` and `contents`. |
-| [Space](/uilib/layout/space/properties) | Various  | _(optional)_ Spacing properties like `top` or `bottom` are supported.            |
+| Property                                | Type     | Description                                                                                                    |
+| --------------------------------------- | -------- | -------------------------------------------------------------------------------------------------------------- |
+| `help`                                  | `object` | _(optional)_ Provide a help button. Object consisting of `title` and `contents`.                               |
+| `countries`                             | `string` | _(optional)_ List only a certain set of countries: `Scandinavia`, `NorthernNordic`, `Europe` or `Prioritized`. |
```
[the remark this answers is not in the document]
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/SelectCountry/properties.mdx]: Yes.

**Comment by langz** [comment]
Done some quick testing at https://eufemia-git-feat-select-country-autocomplete-eufemia.vercel.app/uilib/extensions/forms/feature-fields/SelectCountry/demos/ :

Could the "autocomplete/dropdown list" match the width of the "input field"?

https://github.com/dnbexperience/eufemia/assets/1359205/bd3ad04f-5d94-43e2-bf2c-873fc80082ae

**Comment by langz** [comment]
Done some quick testing at https://eufemia-git-feat-select-country-autocomplete-eufemia.vercel.app/uilib/extensions/forms/feature-fields/SelectCountry/demos/ :

Seems like the search functionality doesn't work as expected, when searching for "ch" I get a lot of results that I don't understand why I get.

https://github.com/dnbexperience/eufemia/assets/1359205/6a35750d-3d92-4958-aece-7039d6e062df

**Comment by tujoworker** [comment]
Nice findings! Regarding the `ch` is because we add the English names too. But I think we should then remove it, and use a different approach regarding the autofill feature 👍

**Comment by langz** [comment]
> Nice findings! Regarding the `ch` is because we add the English names too. But I think we should then remove it, and use a different approach regarding the autofill feature 👍
> 
> And regarding "saudi-arabia" – it should still filter as before. You may press enter or down arrow to open it again.

In regards to the 'ch'-issue, could we only search in the locale used?
So when searching for "ch" in norwegian locale, we would not get "china" as a result?

**Comment by tujoworker** [comment]
yes. The latest commit does ensure that 👍

**Comment by langz** [comment]
> yes. The latest commit does ensure that 👍

Great 👏 
I'll try to test it once more then 🏎️

**Comment by langz** [comment]
Is this a potential issue, do we need to handle if the value is a the ISO code of a country that's not in the given countries value? Or maybe I've just created a strange case that we don't have to handle or provide support for?

Like this:
```
  <Field.SelectCountry
      countries="Nordic"
      value="AL" //Albania(not in the Nordics)
      onChange={(value) => console.log('onChange', value)}
/>
```



[CSB](https://codesandbox.io/p/sandbox/eufemia-starter-forked-rlmds7?file=%2Fsrc%2FApp.tsx%3A7%2C16)

**Pull request merged**